### PR TITLE
Remove max buf size

### LINKED ISSUE #13: Remove MAX_BUF_SIZE limitations
I'm currently working on a AMQP client, the maximum theoretical size for a frame is 2^32-- slightly over what netbuf allows.

I was thinking of writing a fix, but am wondering on what would be the most acceptable solution.

1. 2 structures, Buf and UnboundBuf-- UnboundBuf does most of the work, but Buf ensures that the capacity remains less than MAX_BUF_SIZE.
2. Like above, but instead of hard-coding the maximum buffer size, have it configurable.
3. 1 structure, but the functionality of MAX_BUF_SIZE toggled off with a feature (or toggled on by removing a default feature).

Thoughts?

### PR DESCRIPTION
**Pull request by alreece45**
Changes the capacity to usize and removes MAX_BUF_SIZE.

The latter makes this a breaking change.

Closes #13

### CODE REVIEW

File: src/buf.rs [hunk 1]
```diff
@@ -73,30 +67,29 @@ impl Buf {
 
             if self.consumed() > 0 { // let's allocate new slice and move
                 let min_size = old_bytes + bytes;
-                assert!(min_size < MAX_BUF_SIZE);
-                let mut vec = Vec::with_capacity(
-                    max(min_size, min(old_cap*2, MAX_BUF_SIZE)));
+
+                let mut vec = Vec::with_capacity(max(min_size, old_cap*2));
```
Comment (tailhook) [inline src/buf.rs]: `old_cap*2` should be a `old_cap.saturating_mul(2)` because on 32bit platforms it can easily overflow.

**Review by tailhook — COMMENTED** [review]
Looks good except that one obvious overflow.

I'll take another look over all the code to make sure that no overflow occurs in other places due to increasing maximum buffer.

**Comment by tailhook** [comment]
Okay. Looks good. Thank you!

I'll also take some time to search through all the to do issues I have in case I had another tiny incompatible change in mind. But nothing big is going to be changed here. And then will release to crates.io. Hopefully you're not in hurry?

**Comment by alreece45** [comment]
@tailhook No hurry-- would prefer to make sure things are done right

**Pull request merged**